xarray/core/dataset.py

```diff
@@ -2974,7 +2992,11 @@ def unstack(self, dim=None):
             result = result._unstack_once(dim)
         return result
 
-    def update(self, other, inplace=None):
+    def update(
+        self,
+        other: 'Union[Dataset, Mapping[Hashable, DataArray]]',
```

Comment:
unsure about this - are there more options available for 'other'?

Reply:
This goes through the `merge` machinery (same as `Dataset.__init__`), so in principle I think you can put `Variable` objects or tuples of Variable arguments in the mapping values, too.